jdaviz/app.py

```diff
@@ -1006,6 +1006,17 @@ def return_unique_name(self, data_label, ext=None):
         if number_of_duplicates > 0:
             data_label = f"{data_label} ({number_of_duplicates})"
 
+        # It is possible to add data named "test (1)" and then
+        # add another data named "test" and return_unique_name will see the
+        # first test and assume the second is the duplicate, appending
+        # "(1)" to the end. This overwrites the original data and
+        # causes issues. This block appends a timestamp if a duplicate
+        # is still found in data_collection.
+        if data_label in self.data_collection.labels:
+            import time
+            ts = time.time()
+            data_label = f"{data_label} timestamp: {ts}"
```

Comment:
`time.time()` is not user-friendly. Should format it at least to be more readable.

Reply:
This covers what is in my opinion a rare use case. A user adds data with a label plus "(x)" at the end where x is a number, and then they add data with the exact same label minus the "(x)" at the end. The timestamp would only be added in that one case, which again, I do not foresee happening (often). If I'm wrong in that assumption, please correct me! The alternative to appending a unique string would be data in data collection getting overwritten and causing all kinds of problems. That said, I can try to think of a different way to handle this edge case that preserves the standard naming convention.

I can look into a different method to `time.time()` but do you know one off the top of your head you would prefer to see?

Reply:
At least make the timestamp human-readable using https://docs.python.org/3/library/time.html#time.strftime ?

If the labeling following a simple rule, I still think it is possible to look for all the `label (n)` matches, strip `n` out and then create a new non-conflicting label that is `n + 1`. 

I vaguely remember implementing something like that for Imviz at one point but it was removed as part of your PR to do the labeling stuff for the entire Jdaviz. I used something like this:

https://github.com/spacetelescope/jdaviz/blob/60fc1bdc039d6e1f60c5c9027ed8c8a67bcc9b7f/jdaviz/core/helpers.py#L709

Reply:
Latest commit implements what you wanted here. For example:
```
add_data("test (1)")
add_data("test (47)")
add_data("test")
```

The last `add_data` call will now set the label to be "test (48)".

Reply:
I think this could still be confusing behavior, but it's definitely an improvement and I agree it's an edge case that shouldn't happen too often so should be fine for now.

Reply:
What would you expect to happen in that example?